hydrolib/core/io/bc/models.py

```diff
@@ -236,6 +291,152 @@ class T3D(ForcingBase):
     _verticalpositiontype_validator = get_enum_validator(
         "verticalpositiontype", enum=VerticalPositionType
     )
+    _timeinterpolation_validator = get_enum_validator(
+        "timeinterpolation", enum=TimeInterpolation
+    )
+
+    @root_validator(pre=True)
+    def _validate_quantityunitpair(cls, values):
+        super()._validate_quantityunitpair(values)
+
+        quantityunitpairkey = "quantityunitpair"
+        quantityunitpairs = values[quantityunitpairkey]
+
+        T3D._validate_that_first_unit_is_time_and_has_no_verticalpositionindex(
+            quantityunitpairs
+        )
+
+        verticalpositionindexes = values.get("verticalpositionindex")
+        number_of_verticalpositions = len(values["verticalpositions"])
+
+        if verticalpositionindexes is None:
+            T3D._validate_that_all_non_time_quantityunitpairs_have_valid_verticalpositionindex(
+                quantityunitpairs, number_of_verticalpositions
+            )
+            return values
+
+        T3D._validate_verticalpositionindexes_and_update_quantityunitpairs(
```

Comment:
For clarity maybe put this in an `else` clause? (now it really relies on the previous `return` to not come here.)

Reply:
The `else` clause will add unnecesarry indentation to our code. So I prefer not to add the `else` clause.